sourcegraph/sourcegraph
# JetBrains: refine embedding status indicator

Previously, the embedding status indicator had a lot of gaps with how it works in VS Code. This PR closes many such gaps:

- New icon for "missing embedding"
- New explanation "No file selected" when no file is open
- New explanation "Repository $NAME is not indexed and is missing embeddings" instead of "Repository does not exist on Sourcegraph"

The diff is large because I converted most of the classes to Kotlin while working on this.

## Test plan

- Open a repository that has no embeddings on sorucegraph like https://github.com/fesh0r/fernflower
- Before opening a file, confirm it says "No file selected"
<img width="201" alt="CleanShot 2023-09-14 at 13 29 36@2x" src="https://github.com/sourcegraph/sourcegraph/assets/1408093/e5f72ac7-c0b8-44fd-ad04-d382b6d9ebef">

- After opening a file, confirm it has this icon 
<img width="560" alt="CleanShot 2023-09-14 at 13 29 15@2x" src="https://github.com/sourcegraph/sourcegraph/assets/1408093/7309b41f-5f2c-47bb-b304-73183a1e6bb1">

- Navigate to an external dependency like `java.lang.String`, confirm that the repository is still fernflower (previously it switched to no repository)
